pkg/pulumiyaml/run.go

```diff
@@ -146,6 +146,76 @@ func LoadDir(directory string) (*ast.TemplateDecl, syntax.Diagnostics, error) {
 	return template, diags, nil
 }
 
+// Load a plugin template from the given directory.
+func LoadPluginTemplate(directory string) (*ast.TemplateDecl, syntax.Diagnostics, error) {
+	// Get all yaml files in the directory, load them and merge them into a single template.
```

Comment:
Are we sure we want this now? Are we sure we want to merge all YAML files? How will it work for YAML programs and their stack files, deployment setting files, arbitrary YAML files that users left in their folders, etc?

My preference is probably to keep it to a single file for now, and figure out merging in a separate PR later on the way it would work for any YAML program, not just plugins. I don't feel like we need multi-file for the launch.

Reply:
This is just merging for yaml plugins, that's all new so I'd like to have this. We can work out what we want to do for normal programs later, but it would be good to make them merge as well.

Reply:
Why do you want to have it? Are we going to recommend users create a file per component? I'd keep it simple and show just PulumiPlugin.yaml in all our docs.

Reply:
Yeah, I think not merging actually sets us up for more headaches later if we don't do it now.  

> How will it work for YAML programs and their stack files, deployment setting files, arbitrary YAML files that users left in their folders, etc?

Since we're doing this only for plugins and that's new as Fraser mentioned, we don't have to worry about this, because nobody can expect this to work otherwise.  And I think merging yaml files is neat, I remember splitting up YAML files being requested multiple times.

Reply:
Here's the issue requesting this: https://github.com/pulumi/pulumi-yaml/issues/227.  It's the most upvoted issue in this repo with a bunch of comments.  And since we can support it without having to worry about backwards compat now, I think we should just do it.